themes/bootstrap3/templates/record/addrating.phtml

```diff
@@ -0,0 +1,23 @@
+<?php
+    // Set page title.
+    $this->headTitle($this->translate('rating_add_or_update'));
+
+    // Set up breadcrumbs:
+    $this->layout()->breadcrumbs = $this->searchMemory()->getLastSearchLink($this->transEsc('Search'), '<li>', '</li> ')
+      . '<li>' . $this->recordLinker()->getBreadcrumbHtml($this->driver) . '</li> '
+      . '<li class="active">' . $this->transEsc('rating_add_or_update') . '</li>';
+?>
+<h2><?=$this->transEsc('rating_add_or_update') ?></h2>
+<form method="post" name="rateRecord" class="form-add-rating">
+  <input type="hidden" name="submit" value="1" />
+  <input type="hidden" name="id" value="<?=$this->escapeHtmlAttr($this->driver->getUniqueId())?>" />
+  <input type="hidden" name="source" value="<?=$this->escapeHtmlAttr($this->driver->getSourceIdentifier())?>" />
+  <div class="form-group">
+    <label class="control-label" for="add_rating"><?=$this->transEsc('rating_prompt')?>:</label>
+    <?=$this->render('Helpers/star-rating.phtml', ['ratingData' => $this->currentRating, 'inputAttrs' => ['data-submit-on-change']]);?>
+  </div>
+  <div class="form-group">
+    <input class="btn btn-primary" type="submit" value="<?=$this->transEscAttr('Save')?>"/>
+  </div>
+</form>
+<?php
```

Comment:
Is there a reason for this random open tag at the end?

Reply:
No, removed.